src/librustc_back/target/wasm32_unknown_emscripten.rs

```diff
@@ -18,6 +18,8 @@ pub fn target() -> Result<Target, String> {
                           vec!["-s".to_string(),
                                "BINARYEN=1".to_string(),
                                "-s".to_string(),
+                               "BINARYEN_METHOD='native-wasm,interpret-binary'".to_string(),
```

Comment:
interpret-binary is not necessary any more, node 8.0 can run wasm. See https://github.com/rust-lang/rust/pull/42631